addons/hr_payroll/models/hr_payslip.py

```diff
@@ -0,0 +1,581 @@
+                    blacklist += [id for id, seq in self.env['hr.salary.rule']._recursive_search_of_rules([rule])]
+
+        result = [value for code, value in result_dict.items()]
+        return result
+
+    @api.multi
+    @api.onchange('employee_id', 'date_from')
+    def onchange_employee_id_wrapper(self):
+        self.ensure_one()
+        values = self.onchange_employee_id(self.date_from, self.date_to, self.employee_id.id, self.contract_id.id)['value']
+        for fname, value in values.iteritems():
+            setattr(self, fname, value)
+
+    @api.multi
+    def onchange_employee_id(self, date_from, date_to, employee_id=False, contract_id=False):
+        HrEmployee = self.env['hr.employee']
+        HrPayslipWorkedDays = self.env['hr.payslip.worked_days']
+        HrPayslipInput = self.env['hr.payslip.input']
+
+        if self.env.context is None:
+            self.env.context = {}
+        #delete old worked days lines
+        old_worked_days_ids = self.ids and HrPayslipWorkedDays.search([('payslip_id', '=', self.ids[0])]) or False
+        if old_worked_days_ids:
+            old_worked_days_ids.unlink()
+
+        #delete old input lines
+        old_input_ids = self.ids and HrPayslipInput.search([('payslip_id', '=', self.ids[0])]) or False
+        if old_input_ids:
+            old_input_ids.unlink()
+
+        #defaults
+        res = {'value':{
+                      'line_ids':[],
+                      'input_line_ids': [],
+                      'worked_days_line_ids': [],
+                      #'details_by_salary_head':[], TODO put me back
+                      'name':'',
+                      'contract_id': False,
+                      'struct_id': False,
+                      }
+            }
+
+        if (not employee_id) or (not date_from) or (not date_to):
+            return res
+        ttyme = datetime.fromtimestamp(time.mktime(time.strptime(date_from, "%Y-%m-%d")))
+        employee_id = HrEmployee.browse(employee_id)
+        res['value'].update({
+                    'name': _('Salary Slip of %s for %s') % (employee_id.name, tools.ustr(ttyme.strftime('%B-%Y'))),
+                    'company_id': employee_id.company_id.id
+        })
+
+        if not self.env.context.get('contract', False):
+            #fill with the first contract of the employee
+            contract_ids = self.get_contract(employee_id, date_from, date_to)
+        else:
+            if contract_id:
+                #set the list of contract for which the input have to be filled
+                contract_ids = [contract_id]
+            else:
+                #if we don't give the contract, then the input to fill should be for all current contracts of the employee
+                contract_ids = self.get_contract(employee_id, date_from, date_to)
+
+        if not contract_ids:
+            return res
+        contract_record = self.env['hr.contract'].browse(contract_ids[0])
```

Comment:
why browse with contract_ids[0]? simply not enough self.env['hr.contract'].browse(contract_ids)?